### PR TITLE
 feat: [CDS-76194]: adding manifestInfo in StepInfo for Api usage

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
adding Manifest Info into ServiceExecutionSummary
## Testing
Harness store:
Screenshot 2023-08-23 at 9 38 44 AM
GitStore
Screenshot 2023-08-23 at 9 27 56 AM
Custom
Screenshot 2023-08-23 at 9 32 06 AM
Tested with APi
Screenshot 2023-08-23 at 11 36 14 AM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51945)

### CODE REVIEW

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGModuleInfoProvider.java [hunk 1]
```diff
@@ -105,7 +121,54 @@ public ArtifactsSummary mapArtifactsOutcomeToSummary(Optional<ArtifactsOutcome>
...
+      K8sManifestOutcome k8sManifestOutcome = (K8sManifestOutcome) manifestOutcome;
+      manifestInfoBuilder.valuesPaths(getParameterFieldValue(k8sManifestOutcome.getValuesPaths()));
+    } else if (ManifestType.HelmChart.equals(manifestOutcome.getType())) {
+      HelmChartManifestOutcome helmChartManifestOutcome = (HelmChartManifestOutcome) manifestOutcome;
+      manifestInfoBuilder.chartName(getParameterFieldValue(helmChartManifestOutcome.getChartName()));
+      manifestInfoBuilder.chartVersion(getParameterFieldValue(helmChartManifestOutcome.getChartVersion()));
+      manifestInfoBuilder.helmVersion(helmChartManifestOutcome.getHelmVersion().toString());
+      manifestInfoBuilder.subChartPath(getParameterFieldValue(helmChartManifestOutcome.getSubChartPath()));
+    }
+    // kustomize and openshift what to specify unclear
```
Comment (TarunUba1) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGModuleInfoProvider.java:141]: @namantalaycha10 what all do we need for kustomize and openshift, a bit unclear on it

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGModuleInfoProvider.java [hunk 2]
```diff
@@ -105,7 +121,54 @@ public ArtifactsSummary mapArtifactsOutcomeToSummary(Optional<ArtifactsOutcome>
...
+      manifestInfoBuilder.region(getParameterFieldValue(s3StoreConfig.getRegion()));
+      manifestInfoBuilder.folderPath(getParameterFieldValue(s3StoreConfig.getFolderPath()));
+    } else if (storeConfig instanceof OciHelmChartConfig) {
+      OciHelmChartConfig ociHelmChartConfig = (OciHelmChartConfig) storeConfig;
+      manifestInfoBuilder.folderPath(getParameterFieldValue(ociHelmChartConfig.getBasePath()));
+    } else if (storeConfig instanceof GcsStoreConfig) {
+      GcsStoreConfig gcsStoreConfig = (GcsStoreConfig) storeConfig;
+      manifestInfoBuilder.bucketName(getParameterFieldValue(gcsStoreConfig.getBucketName()));
+      manifestInfoBuilder.folderPath(getParameterFieldValue(gcsStoreConfig.getFolderPath()));
+    } else if (storeConfig instanceof HarnessStore) {
```
Comment (abhinavsingh1196) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGModuleInfoProvider.java:162]: we can get rid of these by introducing new method in interface StoreConfig and override in GitStoreConfig/S3StoreConfig...etc

something like 

`default setManifestStoreInfo(ManifestInfoBuilder builder) {
  return builder.build();
}`


And similarly for ManifestOutcome above?

**Comment by TarunUba1** [comment]
trigger uts

**Review by abhinavsingh1196 — CHANGES_REQUESTED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGModuleInfoProvider.java [hunk 3]
```diff
@@ -105,7 +121,54 @@ public ArtifactsSummary mapArtifactsOutcomeToSummary(Optional<ArtifactsOutcome>
...
     return artifactsSummaryBuilder.build();
   }
-
+  public ManifestInfo mapManifestsOutcomeToSummary(Optional<ManifestsOutcome> manifestOutcomeOptional) {
+    ManifestInfo.ManifestInfoBuilder manifestInfoBuilder = ManifestInfo.builder();
+    if (manifestOutcomeOptional == null || !manifestOutcomeOptional.isPresent()) {
+      return manifestInfoBuilder.build();
+    }
+    List<String> keys = new ArrayList<>(manifestOutcomeOptional.get().keySet());
+    ManifestOutcome manifestOutcome = manifestOutcomeOptional.get().get(keys.get(0));
```
Comment (abosii-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGModuleInfoProvider.java:130]: What if the order is not the one expected? And first is the Values YAML?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Comment by acasian-harness** [comment]
@TarunUba1 the branch name doesn't correspond to jira ticket: https://harness.atlassian.net/browse/CDS-76194

**Pull request closed without merging**